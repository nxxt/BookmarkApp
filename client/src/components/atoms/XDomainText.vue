<template>
  <div class="x-domain-text d-flex align-center">
    <v-icon class="mr-1" x-small>mdi-domain</v-icon>
    <span>{{ text }}</span>
  </div>
</template>

<script lang="ts">
import { Component, Prop, Vue } from 'vue-property-decorator'

@Component
export default class XDomainText extends Vue {
  @Prop({ type: String, default: 'domain' })
  text!: string
}
</script>
