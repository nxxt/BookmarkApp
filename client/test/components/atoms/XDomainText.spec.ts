import Vuetify from 'vuetify'
import { mount, createLocalVue } from '@vue/test-utils'
import XDomainText from '@/components/atoms/XDomainText.vue'

const localVue = createLocalVue()

describe('XDomainText.vue', () => {
  let vuetify: typeof Vuetify
  beforeEach(() => {
    vuetify = new Vuetify()
  })

  const propsData = {
    text: 'test.com'
  }

  const factory = (options?: any) => {
    return mount(XDomainText, {
      localVue,
      vuetify,
      ...options
    })
  }

  it('props', () => {
    const wrapper = factory({ propsData })
    expect(wrapper.props()).toStrictEqual(propsData)
  })

  describe('template parts', () => {
    it('text', () => {
      const wrapper = factory({ propsData })
      const elm = wrapper.find('span')
      expect(elm.text()).toBe(propsData.text)
    })
  })

  describe('the entire template', () => {
    it('snapshot', () => {
      const wrapper = factory({ propsData })
      expect(wrapper.html()).toMatchSnapshot()
    })
  })
})
