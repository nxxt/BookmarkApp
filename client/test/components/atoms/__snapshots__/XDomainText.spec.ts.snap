// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`XDomainText.vue the entire template snapshot 1`] = `"<div class=\\"x-domain-text d-flex align-center\\"><i aria-hidden=\\"true\\" class=\\"v-icon notranslate mr-1 mdi mdi-domain theme--light\\" style=\\"font-size: 12px;\\"></i> <span>test.com</span></div>"`;
